system/Encryption/Handlers/SodiumHandler.php

```diff
@@ -25,10 +25,8 @@ class SodiumHandler extends BaseHandler
 {
     /**
      * Starter key
-     *
-     * @var string
      */
-    protected $key = '';
+    protected ?string $key = null;
```

Comment:
The `null` value should not be allowed here. We already have the property types set in the configuration file: https://github.com/codeigniter4/CodeIgniter4/blob/develop/app/Config/Encryption.php#L24

If we add types here, I would favor adding them for all properties in both handlers. It will be a BC break, but as you said, it will not affect 99.9% of users.

Reply:
The class is not final, add typed property will be bc break

Reply:
I think we should not introduce breaking changes in patch releases.

First of all, why PHPStan reports these errors?

Reply:
The function `sodium_memzero` sets the argument passed to it to `null`.

Reply:
Oh, I got. Thanks!

```
php > $s = 'abc';
php > sodium_memzero($s);
php > var_dump($s);
php shell code:1:
NULL
```

Reply:
Oh... I thought the problem was somewhere else. This changes a lot - thanks.